libs/auth/src/drivers/interfaces/auth-service.interface.ts

```diff
@@ -0,0 +1,13 @@
+import { Observable } from 'rxjs';
+import { DaffAuthToken } from '../../models/auth-token';
+
+export interface DaffAuthServiceInterface<
+  TRequest extends DaffAuthToken,
+> {
+  /**
+   * Checks the validity of an auth token.
+   *
+   * @returns Whether the auth token is valid.
+   */
+  check(auth: TRequest): Observable<boolean>;
```

Comment:
This should not be generic. `check` should probably take no arguments and return void. 

We'll be adding the token to the relevant requests as a header, and we can assume a successful response is true and an error response (403) is false.